common/src/main/java/org/jboss/hal/testsuite/testlistener/ServerStateListenerOperations.java

```diff
@@ -0,0 +1,124 @@
+import static org.jboss.as.controller.client.helpers.ClientConstants.SERVER;
+import static org.jboss.as.controller.client.helpers.ClientConstants.SERVER_CONFIG;
+import static org.jboss.as.controller.client.helpers.ClientConstants.STATUS;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import org.jboss.hal.testsuite.creaper.ManagementClientProvider;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.admin.Administration;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+
+/**
+ * Helper class to read server state informations from model for {@link ServerStateListener}. <br />
+ * Created by pjelinek on Apr 19, 2016
+ */
+public class ServerStateListenerOperations {
+
+    public Map<String, Map<String, String>> getServerStateMap() throws IOException {
+        return tryWithClient(client -> {
+            ModelReader reader = new ModelReader(client);
+            return reader.getHostStream().collect(toMap(identity(), reader.getServerOfHostStatesFunction()));
+        });
+    }
+
+    public boolean isReloadRequired() throws IOException {
+        return tryWithClient(client -> new Administration(client).isReloadRequired());
+    }
+
+    public boolean isRestartRequired() throws IOException {
+        return tryWithClient(client -> new Administration(client).isRestartRequired());
+    }
+
+    private <T> T tryWithClient(Function<OnlineManagementClient, T> modelReadFunction) throws IOException {
+        try (OnlineManagementClient client = ManagementClientProvider.createOnlineManagementClient()) {
+            return modelReadFunction.apply(client);
+        }
+    }
+
+    private class ModelReader {
+
+        private ModelReader(OnlineManagementClient client) {
+            this.ops = new Operations(client);
+        }
+
+        private Operations ops;
+
+        /**
+         * {@link Function} instance to return map of server -> status pairs for given host
+         */
+        private Function<String, Map<String, String>> getServerOfHostStatesFunction() {
+            return host -> {
+                return getServerStream(host).collect(toMap(identity(), server -> {
+                    return readServerConfigStatus(host, server);
+                }));
+            };
+        }
+
+        private Stream<String> getHostStream() {
+            ModelNodeResult hostsResult = null;
+            try {
+                hostsResult = ops.readChildrenNames(Address.root(), HOST);
+            } catch (IOException e) {
+                throw new RuntimeException("Unable to read hosts.", e);
+            }
+            hostsResult.assertDefinedValue();
+            return hostsResult.stringListValue().stream();
```

Comment:
The try-catch block is here just to change type of potential exception to runtime to avoid unnecessary try-catch block inside lambda (making the lambda with most of the business logic hard to read). I don't want to avoid exceptions just want to avoid checked ones to keep it easy to read. This way it's causing no harm and it is easy to read IMO.